NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs

```diff
@@ -202,27 +190,99 @@ private IEnumerable<Entity> SpawnEntitiesStaticly(EntitySpawnPoint entitySpawnPo
+            Entity spawnedEntity = new Entity(entitySpawnPoint.LocalPosition,
+                                              entitySpawnPoint.LocalRotation, 
                                               scale,
                                               techType,
                                               cellLevel,
                                               classId,
                                               true,
                                               deterministicBatchGenerator.NextGuid());
+
+            List<Entity> prefabs = GetEntitiesFromClassId(classId, spawnedEntity, deterministicBatchGenerator);
+
             yield return spawnedEntity;
 
             IEntityBootstrapper bootstrapper;
             if (customBootstrappersByTechType.TryGetValue(techType, out bootstrapper))
             {
                 bootstrapper.Prepare(spawnedEntity, deterministicBatchGenerator);
 
-                foreach(Entity childEntity in spawnedEntity.ChildEntities)
+                for (int i = 0; i < spawnedEntity.ChildEntities.Count - prefabs.Count; i++)
                 {
-                    yield return childEntity;
+                    Entity childEntity = spawnedEntity.ChildEntities[i + prefabs.Count];
                 }
             }
         }
 
+        private IEnumerable<Entity> CreateEntityCellRoot(EntitySpawnPoint entitySpawnPoint, DeterministicBatchGenerator deterministicBatchGenerator)
+        {
+            Entity spawnedEntity = new Entity(entitySpawnPoint.LocalPosition, // CellRoot
+                                              entitySpawnPoint.LocalRotation,
+                                              entitySpawnPoint.LocalScale,
+                                              new TechType("None"),
+                                              entitySpawnPoint.AbsoluteEntityCell.Level,
+                                              entitySpawnPoint.ClassId,
+                                              true,
+                                              deterministicBatchGenerator.NextGuid());
+
+            foreach (EntitySpawnPoint childSpawnPoint in entitySpawnPoint.Children)
+            {
+                if (childSpawnPoint.Density > 0)
+                {
+                    List<UwePrefab> prefabs = prefabFactory.GetPossiblePrefabs(childSpawnPoint.BiomeType);
+                    if (prefabs.Count > 0)
+                    {
+                        spawnedEntity.ChildEntities.AddRange(SpawnEntitiesUsingRandomDistribution(childSpawnPoint, prefabs, deterministicBatchGenerator));
+                    }
+                    else if (childSpawnPoint.ClassId != null)
+                    {
+                        spawnedEntity.ChildEntities.AddRange(SpawnEntitiesStaticly(childSpawnPoint, deterministicBatchGenerator));
+                    }
+                }
+            }
+
+            spawnedEntity.ChildEntities.ForEach(c => c.IsChild = true);
```

Comment:
Does this consider all entities (outside of cell roots) as children?

Reply:
As we havent figured out Batch spawning yea it will just set em all as children of the CellRoot